src/python/pants/backend/project_info/rules/source_file_validator.py

```diff
@@ -69,6 +70,7 @@ def register_options(cls, register):
     register('--detail-level', type=DetailLevel, default=DetailLevel.nonmatching,
              help='How much detail to emit to the console.')
 
+  @memoized_method
```

Comment:
Idle question (non-blocking): Should we be avoiding python-side static memoization and favouring rule-graph memoization? It feels like we'll probably end up leaking a bunch of memory if we're storing stuff on python classes, but also, encouraging this kind of thing to be `@rule`s feels like it violates the spirit of "you mostly just write python".

The specific example of memoizing this in the rule graph would be:
```python
@rule(MultiMatcher, [SourceFileValidation])
def get_multi_matcher(sfv):
  return MultiMatcher(sfv.get_options().config)
```

and then `yield Get(MultiMatcher, SourceFileValidation, sfv)` where you currently call `get_multi_matcher()`

Reply:
Yeah, I don't have a strong opinion either way. Not leaking memory is probably a worthy goal.

Reply:
So, assuming that we fix #6555, I think that in many cases `@memoized` is totally fine... the primary correctness issue that could get involved would be when global or static variables are memoizing things. And in those cases, the global/static is the real issue.